frontend/pages/provider/[provider].tsx

```diff
@@ -201,3 +224,31 @@ const Provider: NextPage = () => {
 };
 
 export default Provider;
+
+export const getStaticPaths = () => ({
+  paths: [{ params: { provider: "aws" } }, { params: { provider: "gcp" } }],
+  fallback: false,
+});
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const { provider: providerName } = context.params as unknown as Params;
+  const sourceFileName = "dbInstance.json";
+  const data = (await import(`../../../data/${sourceFileName}`))
+    .default as DBInstance[];
+  // For SEO, showing the first page is enough. So we only need to
+  // pass the first page of data to the page. Passing the whole large
+  // `data` will make this page twice as large to reduce performance.
```

Comment:
```suggestion
  // `data` will make this page twice as large and reduce performance.
```

Reply:
I'll include this change in the next PR.